File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int maxStreak = 0;

        for(int num : numSet) {
            if(!numSet.count(num - 1)) { // Only start from sequence start
                int currNum = num;
                int currStreak = 1;

                while(numSet.count(currNum + 1)) {
                    currNum++;
                    currStreak++;
                }

                maxStreak = max(maxStreak, currStreak);
            }
        }

        return maxStreak;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.Arrays;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        
        Arrays.sort(nums);
        
        int res = 1;
        int cnt = 1;
        
        for (int i = 1; i < nums.length; i++) {
            if (nums[i-1] == nums[i]) {
                continue;
            }
            
            if (nums[i-1] + 1 == nums[i]) {
                cnt++;
            } else {
                cnt = 1;
            }
            res = Math.max(res, cnt);
        }
        
        return res;
    }
}
